agent/grpc-external/services/peerstream/stream_resources.go

```diff
@@ -331,13 +332,34 @@ func (s *Server) realHandleStream(streamReq HandleStreamRequest) error {
 
 			return nil
 
+		case err := <-errCh:
+			if err == io.EOF {
+				logger.Info("stream ended by peer")
+				return nil
+			}
+			grpcErr, ok := grpcstatus.FromError(err)
+			if ok {
+				// todo: return a sentinel error, don't string match.
+				if grpcErr.Code() == codes.FailedPrecondition && grpcErr.Message() == "cannot establish a peering stream on a follower node" {
+					logger.Debug("connected with a follower", "error", err)
+					return ConnectedWithFollowerErr
```

Comment:
do we want to `status.TrackRecvError` here as well?

Reply:
Design Q: isn't it redundant to error check and return a separate sentinel error? Could this checking happen by the caller and we just return the error as is?

Reply:
a) it's needed to stop the error log and instead use a debug log b) totally we could just return the error but then I thought we're leaking our protobufs too far outside of this method

Reply:
I ended up returning the error 😄